HLTrigger/HLTcore/interface/TriggerExpressionParser.h

```diff
@@ -37,16 +48,24 @@ namespace triggerExpression {
 
       operand %= (prescale | element);
 
-      unary = ((qi::lit("NOT") >> operand)[qi::_val = new_<OperatorNot>(qi::_1)] | operand[qi::_val = qi::_1]);
+      unary = ((operand_not >> operand)[qi::_val = new_<OperatorNot>(qi::_1)] | operand[qi::_val = qi::_1]);
```

Comment:
```suggestion
      unary = ((operand_not >> expression)[qi::_val = new_<OperatorNot>(qi::_1)] | operand[qi::_val = qi::_1]);
```
Playing around, it seemed like this could fix the "NOT NOT [expr]" case.

Reply:
Ah... non sapevo si potesso usare rules ricorsive

Reply:
Se funziona, fai un altro commit (credo ci sia anche un test da cambiare) :+1:

Reply:
Okay, allora intanto faccio il merge di questa, poi ricontrollo il tutto.

Reply:
> Playing around, it seemed like this could fix the "NOT NOT [expr]" case.

https://github.com/missirol/cmssw/commit/17ffd443196392da9baec10ef779bfa246fe420a contiene questo update, ma ha un limite: cambia il parsing di `NOT THIS OR THAT` in `NOT (THIS OR THAT)`, mentre al momento sarebbe `(NOT THIS) OR THAT`. Forse e' meglio non cambiare quest'ultimo, che e' piu' intuitivo, solo per avere la possibilita' di scrivere `NOT NOT` invece di `NOT (NOT (`, dato che `NOT NOT` sara' raramente usato.

Reply:
No, infatti: direi di che l'interpretazione che ci si aspetta sia `(NOT THIS) OR THAT`.

Reply:
Okay, quindi `NOT THIS OR THAT` rimarra' `(NOT THIS) OR THAT`.

Ora sto provando
```diff
-      unary = ((operand_not >> operand)[qi::_val = new_<OperatorNot>(qi::_1)] | operand[qi::_val = qi::_1]);
+      unary = ((operand_not >> unary)[qi::_val = new_<OperatorNot>(qi::_1)] | operand[qi::_val = qi::_1]);
```
(che, pensadoci, ha piu' senso) e sembra funzionare correttamente.

Reply:
OK, questo mi sembra sensato.